en/debug/porting-guide.md

```diff
@@ -2,55 +2,67 @@
 
 This topic is for developers who want to port PX4 to work with *new* flight controller hardware.
 
-## Architecture
+## PX4 Architecture
 
-PX4 consists of two main layers: The board support and middleware layer on top of the host OS (NuttX, Linux or any other POSIX platform like Mac OS). And the applications (Flight Stack in [src/modules](https://github.com/PX4/Firmware/tree/master/src/modules)\).
+PX4 consists of two main layers: The [board support and middleware layer](https://dev.px4.io/en/middleware/) on top of the host OS (NuttX, Linux or any other POSIX platform like Mac OS), and the applications (Flight Stack in [src/modules](https://github.com/PX4/Firmware/tree/master/src/modules)\).  Please reference the [PX4 Architectural Overview](../concept/architecture.md) for more information.
```

Comment:
Links within the same doc should be relative. So for example this `https://dev.px4.io/en/middleware/` should be `../middleware/README.md`

Reply:
Ah, ok, so anything that's in the "Devguide" repo should use a relative link, I should only use a full URL if I need to reference something else, e.g. the User Guide?  I was working through the GitHub interface to make this update and was a little unsure of how it handled things.  I will update this.

Reply:
That is correct. You've done it right throughout pretty much - except for this one.